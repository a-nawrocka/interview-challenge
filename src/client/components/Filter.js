import React from "react";

class Filter extends React.Component {
	onFilter = (term) =>  {
		this.props.onFilter(term);		
	};

	render() {
		return (
			<div className="filters">
              <input className="form-control" placeholder="Name" onChange={ e =>this.onFilter(e.target.value.toLowerCase())} />
            </div>
		);
	};
}

export default Filter;