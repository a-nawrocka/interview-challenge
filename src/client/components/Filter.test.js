import React from "react";
import renderer from "react-test-renderer";
import Filter from "./Filter";
import { shallow } from "enzyme";

describe("Given onFilter function", () => {
	it("display filter input box", () => {
		const component = renderer.create(<Filter onFilter={() => {}} />);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe("When filter is changed", () => {
		it("applies the filter", () => {
			const mockOnFilter = jest.fn();

			const filterComponent = shallow(<Filter onFilter={mockOnFilter} />);
			filterComponent.find("input").simulate("change", { target: { value: "Kale" } });

			expect(mockOnFilter.mock.calls.length).toBe(1);
			expect(mockOnFilter.mock.calls[0][0]).toBe("kale");
		});
	});
});
