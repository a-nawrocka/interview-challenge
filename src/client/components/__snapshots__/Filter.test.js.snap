// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given onFilter function display filter input box 1`] = `
<div
  className="filters"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Name"
  />
</div>
`;
